[ROCM] OPT Pass plugin test refactoring and organization

josemonsalve2 opened the pull request:
Moving the test, renaming, and cleaning. 

We are moving the test from `compiler/plugins/target/ROCM/test/` to `samples/compiler_plugins/rocm/` to make it more in line with the organization of the project. 

This also changes the name of the test from `GPUHello` to `ROCMPluginTest`.

This also cleans up the `CMakeLists.txt` file for this test.

This is to solve the issues discussed in #18347

josemonsalve2 committed:
Moving test to samples. Changing test name from GPUHello to ROCMPluginTest, Fixing CMakeLists.txt

Signed-off-by: Jose M Monsalve Diaz <jmonsalv@amd.com>

ScottTodd commented on samples/compiler_plugins/rocm/opt_pass_plugin/ROCMPluginTest.cpp:
How about "Sample" or "Demo" instead of "Test"?

ScottTodd reviewed (commented):
Thanks! Please sync past the merge conflict so the github workflows run.

Also FYI, I'll be traveling next week and won't have my Windows dev machine to help with testing.

josemonsalve2 committed:
Renaming test from ROCMPluginTest to ROCMPluginSample

Signed-off-by: Jose M Monsalve Diaz <jmonsalv@amd.com>

josemonsalve2 committed:
Adding condition to only run test when ROCM backend is built.

Otherwise, the test will fail because the rocm specific flag is not recognized.

Signed-off-by: Jose M Monsalve Diaz <jmonsalv@amd.com>
